### DISCUSSION TITLE
LAB #A Ch6 Host Code Optimization fail to build the Device Binary(xclbin)

### DISCUSSION THREAD

**Discussion by s095339** [bol-edu/HLS-SOC-Discussions#39] [Q&A] [answered]
Hi, We have been doing Ch6 in Lab A. The LabName is "Host Code Optimization"   
We rent a u50 device and connected to the board and followed the tutorial and access to 
"Hardware_Acceleration/Design_Tutorials/07-host-code-opt/reference-files"
Then  typed  
"make TARGET=hw DEVICE=xilinx_u50_gen3x16_xdma_5_202210_1 xclbin" 
in commandline to build the Device, but we got a few error as shown below.

![image](https://user-images.githubusercontent.com/78518884/223767431-58d900cb-0f32-4232-879c-4a06dba2e63e.png)
I have no idea how to solve the problem, Is there anything we need to modify before building? Thank you.

**Comment by ChihyuLin0211** [comment, marked as answer]
Hi, 

Since Alevo U50 doesn't have DDR memory resource, you need to specify things a bit differently.
![image](https://user-images.githubusercontent.com/50320654/223909318-80040275-6f36-4b46-b3e3-853a97374a1a.png)

You may use the platforminfo command to query the resources for a specific platform or shell. 
$platforminfo -p xilinx_u50_gen3x16_xdma_5_202210_1

**Reply by EvanLu0815** [reply]
You need to complete only Co-sim for design analysis in Lab A.

**Comment by s095339** [comment]
@EvanLu0815 @ChihyuLin0211 Thanks!! I've got everything done correctly!!

Let me share my solution to anyone who are going to choose this topic in the future.

The code is originally configured for U200 platform, but I'm using U50 for this Lab.
Since the U50 platform doesn't have DDR, I got error when compiling  the whole project.
![image](https://user-images.githubusercontent.com/78518884/225714143-b9131fff-b6eb-4155-a3e9-d9a6cb8611a1.png)

The solution is:
First, head to the design.cfg file under the reference-files dir and change the DDR[0] DDR[1] to HBM[0] HBM[1].
![image](https://user-images.githubusercontent.com/78518884/225712903-147508da-e898-4ffd-9d96-92d25fc0e7a3.png)

Then, go to Makefile. Modify the DEVICE name to xilinx_u50_gen3X16_xdma_5_202210_1 ( It depends on what platform you're using)
![image](https://user-images.githubusercontent.com/78518884/225713350-90cc2ce1-7091-454d-b850-47b073ae8281.png)

Hope this can helps someone who also choose this topic.

**Reply by jiinlai** [reply]
Good. Thank you for sharing.
